pkg/aggregator/ckey/key.go

```diff
@@ -27,49 +26,39 @@ import (
 // Note that Agent <= 6.19.0 were using a 128 bits hash, we've switched
 // to 64 bits for better performances (map access) and because 128 bits were overkill
 // in the first place.
-// Note that we've benchmarked against xxhash64 which should be slightly faster,
-// but the Go compiler is not inlining xxhash sum methods whereas it is inlining
-// the murmur3 implementation, providing better performances overall.
 // Note that benchmarks against fnv1a did not provide better performances (no inlining).
 type ContextKey uint64
 
 // KeyGenerator generates key
 // Not safe for concurrent usage
 type KeyGenerator struct {
-	buf []byte
+	intb uint64
 }
 
 // NewKeyGenerator creates a new key generator
 func NewKeyGenerator() *KeyGenerator {
-	return &KeyGenerator{
-		buf: make([]byte, 0, 1024),
-	}
+	return &KeyGenerator{}
 }
 
 // Generate returns the ContextKey hash for the given parameters.
 // The tags array is sorted in place to avoid heap allocations.
 func (g *KeyGenerator) Generate(name, hostname string, tags []string) ContextKey {
-	g.buf = g.buf[:0]
 
-	// Sort the tags in place. For typical tag slices, we use
-	// the in-place insertion sort to avoid heap allocations.
-	// We default to stdlib's sort package for longer slices.
-	// See `pkg/util/sort.go` for info on the threshold.
-	if len(tags) < util.InsertionSortThreshold {
-		util.InsertionSort(tags)
-	} else {
+	// Ensure the list of tags is deduplicated the fastest way possible
+	if len(tags) > util.InsertionSortThreshold {
 		sort.Strings(tags)
+		util.UniqSorted(tags)
+	} else {
+		util.DedupInPlace(tags)
 	}
 
-	g.buf = append(g.buf, name...)
-	g.buf = append(g.buf, ',')
-	for i := 0; i < len(tags); i++ {
-		g.buf = append(g.buf, tags[i]...)
-		g.buf = append(g.buf, ',')
+	g.intb = 0xc6a4a7935bd1e995
```

Comment:
Can we add a comment on this number ?

Reply:
that would be indeed a good thing haha, adding this